podium-gateway/src/main/java/nl/thehyve/podium/service/DeliveryService.java

```diff
@@ -0,0 +1,232 @@
+     * Checks if the delivery process has the required status.
+     * @param deliveryProcess the delivery process object.
+     * @param status the required status.
+     * @throws ActionNotAllowedInStatus iff the delivery process does not have the required status.
+     */
+    private void checkDeliveryStatus(DeliveryProcess deliveryProcess, DeliveryStatus status) throws ActionNotAllowedInStatus {
+        if (deliveryProcess.getStatus() != status) {
+            throw ActionNotAllowedInStatus.forStatus(deliveryProcess.getStatus());
+        }
+    }
+
+    /**
+     * Gets the delivery process with the provided uuid from the request.
+     * @param request the request to get the delivery process for.
+     * @param deliveryProcessUuid the uuid of the delivery process.
+     * @return the delivery process object.
+     * @throws ResourceNotFound iff no delivery process with the uuid exists for the request.
+     */
+    private DeliveryProcess getDeliveryProcess(Request request, UUID deliveryProcessUuid) {
+        Optional<DeliveryProcess> deliveryProcessOptional = request.getDeliveryProcesses().stream()
+            .filter(process -> process.getUuid().equals(deliveryProcessUuid)).findAny();
+        if (deliveryProcessOptional.isPresent()) {
+            return deliveryProcessOptional.get();
+        }
+        throw new ResourceNotFound("Could not find delivery process with uuid " + deliveryProcessUuid.toString());
+    }
+
+    /**
+     *
+     * @param uuid
+     * @return
+     * @throws ActionNotAllowedInStatus
+     */
+    public List<DeliveryProcessRepresentation> getDeliveriesForRequest(UUID uuid) throws ActionNotAllowedInStatus {
+        Request request = requestRepository.findOneByUuid(uuid);
+        checkStatus(request, RequestStatus.Delivery);
+        return request.getDeliveryProcesses().stream()
+            .map(deliveryProcessMapper::deliveryProcessToDeliveryProcessRepresentation)
+            .collect(Collectors.toList());
+    }
+
+    /**
+     * Start delivery processes for each of the selected request types for a request.
+     * @param user the current user.
+     * @param uuid the uuid of the request.
+     * @return the list of generated delivery process instances.
+     * @throws ActionNotAllowedInStatus iff the request is not in status Approved.
+     */
+    public List<DeliveryProcessRepresentation> startDelivery(AuthenticatedUser user, UUID uuid) throws ActionNotAllowedInStatus {
+        Request request = requestRepository.findOneByUuid(uuid);
+        checkStatus(request, RequestStatus.Approved);
+        List<DeliveryProcessRepresentation> deliveryProcesses = new ArrayList<>();
+        for(RequestType type: request.getRequestDetail().getRequestType()) {
+            DeliveryProcess deliveryProcess = deliveryProcessService.start(user, type);
+            request.addDeliveryProcess(deliveryProcess);
+            publishDeliveryStatusUpdate(user, RequestStatus.Approved, request, deliveryProcess, null);
```

Comment:
We could make the sourceStatus dynamic (for all events) using request.getStatus().